### DISCUSSION TITLE
Having trouble connecting to mobile wallets while using web3modal

### DISCUSSION THREAD

**Discussion by adityachakra16** [WalletConnect/walletconnect-monorepo#3296] [Developer Support] [answered]
I have tested with both Rainbow and Trust wallet on both Android and iOS. Issue seems to persist for all of them.

For context, this is a Discourse plugin. 

The code can be found [here](https://github.com/spruceid/discourse-siwe-auth/blob/main/assets/javascripts/discourse/lib/web3modal.js.es6#L11).

There are a few live Discourse forums where this is in use. Here is the [ENS Forum](https://discuss.ens.domains/discourse-siwe/auth) where you can try it out.

**Comment by glitch-txs** [comment, marked as answer]
line 22 of the code, where it says `version: 1`, should be replaced with `version: 2`. Old bridge no longer works 👍

**Reply by adityachakra16** [reply]
I actually thought this was the issue initially. However, when i change it to `version: 2` or dont add version at all, I get the following error.

`Error: No matching key. keychain: <some key>`

After making this change, the web3 modal stops showing up completely.

**Reply by glitch-txs** [reply]
Update to the latest version please

**Reply by glitch-txs** [reply]
You actually don't need to set a version anymore :)

https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/installation

**Reply by adityachakra16** [reply]
Here are the versions I'm currently using.

    "@wagmi/core": "^1.3.3",
    "@web3modal/ethereum": "^2.7.1",
    "@web3modal/html": "^2.7.1",
    "viem": "^1.2.5"

**Reply by adityachakra16** [reply]
Using these versions gives me the `Error: No matching key. keychain: <some key>` error. And the walletconnect web3modal doesnt show up at all.

As I also added previously, I'm not setting `version: 1` anymore.

**Reply by glitch-txs** [reply]
hey, feel free to check our web3modal examples [here](https://github.com/WalletConnect/web3modal-examples).
We also have labs for testing: 
- repo https://github.com/WalletConnect/web3modal
- website: https://lab.web3modal.com/

Happy to help if you could provide a minimal reproducible example, so we don't have to go through the whole source code :)

**Reply by adityachakra16** [reply]
Just found this https://github.com/orgs/WalletConnect/discussions/2728

Seems to be a known issue. I will close this one, thanks
